Code/Engine/GameEngine/XR/Implementation/XRWindow.cpp

```diff
@@ -0,0 +1,171 @@
+#include <GameEnginePCH.h>
+
+#include <GameEngine/XR/XRWindow.h>
+#include <GameEngine/XR/XRInterface.h>
+#include <../../../Data/Base/Shaders/Pipeline/VRCompanionViewConstants.h>
+#include <Core/ResourceManager/ResourceManager.h>
+#include <RendererCore/RenderContext/RenderContext.h>
+#include <RendererFoundation/Device/SwapChain.h>
+#include <RendererFoundation/Resources/Resource.h>
+#include <RendererFoundation/Profiling/Profiling.h>
+#include <RendererFoundation/Resources/Texture.h>
+
+// clang-format off
+EZ_BEGIN_DYNAMIC_REFLECTED_TYPE(ezActorPluginWindowXR, 1, ezRTTINoAllocator);
+EZ_END_DYNAMIC_REFLECTED_TYPE;
+// clang-format on
+
+
+//////////////////////////////////////////////////////////////////////////
+
+ezWindowXR::ezWindowXR(ezXRInterface* pVrInterface, ezUniquePtr<ezWindowBase> pCompanionWindow)
+  : m_pVrInterface(pVrInterface)
+  , m_pCompanionWindow(std::move(pCompanionWindow))
+{}
+
+ezWindowXR::~ezWindowXR() {}
+
+ezSizeU32 ezWindowXR::GetClientAreaSize() const
+{
+  return m_pVrInterface->GetHmdInfo().m_vEyeRenderTargetSize;
+}
+
+ezWindowHandle ezWindowXR::GetNativeWindowHandle() const
+{
+  if (m_pCompanionWindow)
+  {
+    m_pCompanionWindow->GetNativeWindowHandle();
+  }
+  return ezWindowHandle();
+}
+
+bool ezWindowXR::IsFullscreenWindow(bool bOnlyProperFullscreenMode) const
+{
+  //#TODO: return companion state or XR?
```

Comment:
I would expose the companion window as a separate window somehow, so that you can specifically ask for that and modify it. The HMD will always be fullscreen, so just returning true is the right choice.

Reply:
So timing in XR is key and I need full control on when the window is rendered. If both the XR and the companion window are handled by separate ezActors I lose control on the order of execution for timing and dependency purposes (I can only render the companion view as long as I have rendered into the XR texture and am still holding onto that texture.
So I opted for wrapping the companion window into the XR window but that obviously causes some ambiguity. Returning the XR state should be the correct choice in most cases though.